### ISSUE TITLE
Unable to convert between types via numpy.astype 

### ISSUE THREAD

**Issue by Nucs** [SciSharp/NumSharp#280]
Consider the following code:
``` python
a = np.array([1, 2, 3], dtype=np.int32)
a = a.astype(np.float64)
```

The following code in C# does not work:
``` C#
np.array(1, 2, 3, 4, 5).astype(np.float64);
```

Exception: 
```
Test method NumSharp.UnitTest.NpRandomUniformTests.Basic threw exception: 
System.NotImplementedException: _ChangeTypeOfArray from Int32 to Double
    at NumSharp.Backends.TypedArrayStorage._ChangeTypeOfArray(Array arrayVar, Type to_dtype) in C:\Users\Eli-PC\Desktop\NumSharp\src\NumSharp.Core\Backends\TypedArrayStorage.cs:line 135
   at NumSharp.Backends.TypedArrayStorage.SetData(Array values, Type dtype) in C:\Users\Eli-PC\Desktop\NumSharp\src\NumSharp.Core\Backends\TypedArrayStorage.cs:line 718
   at NumSharp.NDArray.astype(Type dtype, Boolean copy) in C:\Users\Eli-PC\Desktop\NumSharp\src\NumSharp.Core\Backends\NDArray.cs:line 132
   at NumSharp.UnitTest.NpRandomUniformTests.Basic() in C:\Users\Eli-PC\Desktop\NumSharp\test\NumSharp.UnitTest\Random\np.random.uniform.Test.cs:line 16
```

**Comment by Nucs** [comment]
I can fix it simply by adding support for all type conversions in `_ChangeTypeOfArray` like it is in python.

**Comment by Oceania2018** [comment]
No, thanks. I’ve already fix that issue on my local.

**Comment by Nucs** [comment]
Was fixed in PR #287
